### PR TITLE
Resolve a number of UI issues

### LINKED ISSUE #510: Scale column isn't updated with unit count
The scale column in the model details view doesn't have any value, it's always '-'. It should have the unit totals.

### PR DESCRIPTION
**Pull request by hatched**
## Done

There were a number of UI issues that needed resolving. See the details list for the full list of issues.

## QA

- Install the juju 2.8-rc2 snap
- `juju bootstrap` into a public cloud
- `juju deploy cs:bundle/canonical-kubernetes`
- `juju dashboard` and connect to that dashboard to accept the self-signed cert.
  - take note of the ip address and u/p
- Open config.js and change: 
  - `baseControllerURL` to the ip address and port (if any) from above.
  - `isJuju` to `true`
  - `identityProviderAvailable` to `false`
- `./run`
- Open http://0.0.0.0:8036/
- Use the above u/p to log in. 
- View the model details for your `default` model which will contain the kubernetes bundle deployment.

## Details

Fixes #510 #511 #512 #513 #514 #516 #518 #519  

## Screenshots
Screen Shot 2020-05-27 at 1 08 25 PM

192 168 64 6_8036_models_default

### CODE REVIEW

**Comment by clagom** [comment]
I leave it to @spencerbygraves as I'm not sure about the typography for the empty tables. Probably aligned to the left? But up to you Spencer.

**Comment by hatched** [comment]
@clagom it's using the same mechanism and styles as the model table empty message. If you'd like to change them all then that can be addressed separately.

File: src/pages/Models/Details/generators.js [hunk 1]
```diff
@@ -57,16 +58,25 @@ export function generateIconImg(name, namespace) {
...
     <>
       {subordinate && <span className="subordinate"></span>}
       {namespace && generateIconImg(name, namespace)}
       {/* Ensure app is not a local charm */}
       {namespace.includes("cs:") ? (
-        <a data-test="app-link" href={`https://www.jaas.ai/${charmStorePath}`}>
+        <a
+          data-test="app-link"
+          target="_blank"
+          rel="noreferrer"
```
Comment (anthonydillon) [inline src/pages/Models/Details/generators.js:77]: Needs `noopener noreferrer`

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "jaas-dashboard",
-  "version": "0.1.4",
+  "version": "0.1.6",
```
Comment (anthonydillon) [inline package.json:3]: Maybe 5?

File: src/components/InfoPanel/InfoPanel.js [hunk 3]
```diff
@@ -24,6 +24,8 @@ const expandedTopologyDimensions = () => {
 
 const infoPanelDimensions = () => {
   const vw = getViewportWidth();
+  // If changes are made to the 1580px breakpoint below then be sure to update
+  // the same breakpoint in _info-panel.scss.
```
Comment (anthonydillon) [inline src/components/InfoPanel/InfoPanel.js:28]: For future but I would connect this with the css via a classname

File: src/components/LogIn/LogIn.js [hunk 4]
```diff
@@ -49,6 +49,9 @@ function generateErrorMessage(loginError) {
     case '"user-" is not a valid user tag':
       loginErrorMessage = "Invalid user name";
       break;
+    case "invalid entity name or password":
+      loginErrorMessage = "invalid user name or password";
```
Comment (anthonydillon) [inline src/components/LogIn/LogIn.js:53]: Should this be capitalised?

File: src/pages/Models/Details/_model-details.scss [hunk 5]
```diff
@@ -7,6 +7,10 @@
   margin-bottom: 2rem;
   padding-bottom: 14rem;
 
+  .p-main-table caption {
```
Comment (anthonydillon) [inline src/pages/Models/Details/_model-details.scss:10]: I don't see a caption on the model details page.

File: src/pages/Models/Details/generators.js [hunk 6]
```diff
@@ -99,10 +108,13 @@ export function generateApplicationRows(
           content: app.status ? generateStatusElement(app.status.status) : "-",
           className: "u-capitalise",
         },
-        { content: "-", className: "u-align--right" },
-        { content: "-", className: "u-align--right" },
+        { content: app.workloadVersion || "-", className: "u-align--right" },
+        { content: Object.keys(app.units).length, className: "u-align--right" },
```
Comment (barrymcgee) [inline src/pages/Models/Details/generators.js:112]: What is `app.units` is not available or not an array?

File: src/pages/Models/Details/generators.js [hunk 7]
```diff
@@ -221,7 +232,10 @@ export function generateMachineRows(modelStatusData, filterByApp) {
         {
           content: (
             <>
-              <div>{machineId}</div>
+              <div>
```
Comment (barrymcgee) [inline src/pages/Models/Details/generators.js:235]: This div negates the need for the fragment wrapper or vice versa.

File: package.json (hunk 2 above)
Comment (hatched) [reply package.json:3]: I was going to go for 9, but yeah 5 is fine :D

File: src/pages/Models/Details/generators.js (hunk 1 above)
Comment (hatched) [reply src/pages/Models/Details/generators.js:77]: done

File: src/components/LogIn/LogIn.js (hunk 4 above)
Comment (hatched) [reply src/components/LogIn/LogIn.js:53]: Yep, done.

File: src/components/InfoPanel/InfoPanel.js (hunk 3 above)
Comment (hatched) [reply src/components/InfoPanel/InfoPanel.js:28]: These are the values used to pass to the topology dimensions so a class won't work as we actually need the numbers.

File: src/pages/Models/Details/_model-details.scss (hunk 5 above)
Comment (hatched) [reply src/pages/Models/Details/_model-details.scss:10]: It's the first screenshot above. You can see it in a controller model if you've deployed your own controller. Or if you create an empty model.

File: src/pages/Models/Details/generators.js (hunk 7 above)
Comment (hatched) [reply src/pages/Models/Details/generators.js:235]: The `<a>` is also part of the element so we still need the fragment.

File: src/pages/Models/Details/generators.js (hunk 6 above)
Comment (hatched) [reply src/pages/Models/Details/generators.js:112]: The application data set always has an empty array for the units as part of the contract but I suppose it's possible that could change. I've made it more resilient.

**Comment by hatched** [comment]
Thanks! @spencerbygraves @clagom if you'd like some additional design/ux updates plz file follow-ups and I'll get right on them.

**Pull request merged**